pkg/datasource/accessors.go

```diff
@@ -322,186 +334,203 @@ func (a *fieldAccessor) Annotations() map[string]string {
+		return 0, ErrFieldTooShort
 	}
-	return a.ds.byteOrder.Uint32(val)
+	return a.ds.byteOrder.Uint32(val), nil
 }
 
-func (a *fieldAccessor) Uint64(data Data) uint64 {
+func (a *fieldAccessor) Uint64(data Data) (uint64, error) {
 	val := a.Get(data)
 	if len(val) < 8 {
-		return 0
+		return 0, ErrFieldTooShort
 	}
-	return a.ds.byteOrder.Uint64(val)
+	return a.ds.byteOrder.Uint64(val), nil
 }
 
-func (a *fieldAccessor) Int8(data Data) int8 {
+func (a *fieldAccessor) Int8(data Data) (int8, error) {
 	val := a.Get(data)
 	if len(val) < 1 {
-		return 0
+		return 0, ErrFieldTooShort
 	}
-	return int8(val[0])
+	return int8(val[0]), nil
 }
 
-func (a *fieldAccessor) Int16(data Data) int16 {
+func (a *fieldAccessor) Int16(data Data) (int16, error) {
 	val := a.Get(data)
 	if len(val) < 2 {
-		return 0
+		return 0, ErrFieldTooShort
 	}
-	return int16(a.ds.byteOrder.Uint16(val))
+	return int16(a.ds.byteOrder.Uint16(val)), nil
 }
 
-func (a *fieldAccessor) Int32(data Data) int32 {
+func (a *fieldAccessor) Int32(data Data) (int32, error) {
 	val := a.Get(data)
 	if len(val) < 4 {
-		return 0
+		return 0, ErrFieldTooShort
 	}
-	return int32(a.ds.byteOrder.Uint32(val))
+	return int32(a.ds.byteOrder.Uint32(val)), nil
 }
 
-func (a *fieldAccessor) Int64(data Data) int64 {
+func (a *fieldAccessor) Int64(data Data) (int64, error) {
 	val := a.Get(data)
 	if len(val) < 8 {
-		return 0
+		return 0, ErrFieldTooShort
 	}
-	return int64(a.ds.byteOrder.Uint64(val))
+	return int64(a.ds.byteOrder.Uint64(val)), nil
 }
 
-func (a *fieldAccessor) Float32(data Data) float32 {
-	return math.Float32frombits(a.Uint32(data))
+func (a *fieldAccessor) Float32(data Data) (float32, error) {
+	i, err := a.Uint32(data)
+	if err != nil {
+		return 0.0, err
+	}
+	return math.Float32frombits(i), nil
 }
 
-func (a *fieldAccessor) Float64(data Data) float64 {
-	return math.Float64frombits(a.Uint64(data))
+func (a *fieldAccessor) Float64(data Data) (float64, error) {
+	i, err := a.Uint64(data)
+	if err != nil {
+		return 0.0, err
+	}
+	return math.Float64frombits(i), nil
 }
 
-func (a *fieldAccessor) String(data Data) string {
+func (a *fieldAccessor) String(data Data) (string, error) {
 	if a.f.Kind == api.Kind_CString {
-		return gadgets.FromCString(a.Get(data))
+		return gadgets.FromCString(a.Get(data)), nil
 	}
-	return string(a.Get(data))
+	return string(a.Get(data)), nil
 }
 
-func (a *fieldAccessor) Bytes(data Data) []byte {
-	return a.Get(data)
+func (a *fieldAccessor) Bytes(data Data) ([]byte, error) {
+	return a.Get(data), nil
 }
 
-func (a *fieldAccessor) Bool(data Data) bool {
+func (a *fieldAccessor) Bool(data Data) (bool, error) {
 	val := a.Get(data)
 	if len(val) < 1 {
-		return false
+		return false, ErrFieldTooShort
 	}
-	return val[0] == 1
+	return val[0] == 1, nil
 }
 
-func (a *fieldAccessor) PutUint8(data Data, val uint8) {
+func (a *fieldAccessor) PutUint8(data Data, val uint8) error {
 	b := a.Get(data)
 	if len(b) < 1 {
-		return
+		return ErrFieldTooShort
```

Comment:
Do you think there are cases when it makes sense to read a Uint8 from something that's larger than 1 byte without specifying an additional offset? Otherwise maybe it's cleaner to just return `ErrInvalidFieldLength` and check for `len(b) != 1`. Also for the other types.

(Usually I'm not into restricting users from doing weird stuff, but here I don't know...)

Reply:
Added that check as well. At this point, I'm wondering if we should just check the type of the field instead... (but I'll host that discussion for a bit).